### PR TITLE
Add personal access tokens for API access

### PR DESCRIPTION
**Pull request by nwalters512**
As a part of #225, adds tokens that can be used to authenticate API requests (based largely on my work in https://github.com/PrairieLearn/PrairieLearn/pull/1260). Users can create, view, and delete tokens on their personal settings page, and they can specify tokens either in a header or as a query parameter.

**TODO**
- [x] Docs

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -35,3 +35,14 @@ described here are located under `/api/`.
 | `/questions/:questionId/answering` | `POST`                 | Mark a question as being answered                   |
 | `/questions/:questionId/answering` | `DELETE`               | Mark a question as no longer being answered         |
 | `/questions/:questionId/answered`  | `POST`                 | Mark the question as answered                       |
+
+## API Access Tokens
+
+When logged into [queue.illinois.edu/q](queue.illinois.edu/q), you're automatically authenticated with the API. However, if you want to access the API from outside the queue to programmatically access or manipulate it, you can create personal access tokens that don't rely on Shibboleth for authentication.
```
Comment (james9909) [inline doc/API.md]: `queue.illinois.edu/q` -> `https://queue.illinois.edu/q`

File: src/middleware/authnJwt.js [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+    // Maybe they authenticated a different way
+    next()
     return
   }
 
   const user = await getUserFromJwt(jwtCookie)
   if (user === null) {
     // This shouldn't ever happen, but if it does...
+    next(new ApiError(401, 'Invalid cookie'))
     res.status(401).send()
```
Comment (james9909) [inline src/middleware/authnJwt.js]:
```suggestion
```

File: src/components/userSettings/AccessTokenListGroupItem.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  return (
+    <ListGroupItem>
+      <div className="d-flex flex-row align-items-center">
+        <div className="d-flex flex-column">
+          <strong>{props.name}</strong>
+          <span className="text-muted">Created at {props.createdAt}</span>
+          <span className="text-muted">
+            {props.lastUsedAt
+              ? `Last used at ${props.lastUsedAt}`
```
Comment (james9909) [inline src/components/userSettings/AccessTokenListGroupItem.jsx]: I'm not too opinionated on this, but maybe we could format the date to be a bit nicer?

Comment (nwalters512) [reply src/components/userSettings/AccessTokenListGroupItem.jsx]: I changed it to show dates the same way we do for questions.

**Pull request merged**